py/src/braintrust/framework.py

```diff
@@ -87,6 +87,21 @@ class EvalScorerArgs(SerializableDataClass):
 ]
 
 
+@dataclasses.dataclass
+class EvalMetadata(SerializableDataClass):
+    """
+    Additional metadata for the eval definition, such as experiment name.
+    """
+
+    experiment_name: Optional[str] = None
```

Comment:
you can write a docstring on this field itself (see how this is done in the summary) too